src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -90,6 +90,37 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
     return err;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetUpTime(uint64_t & upTime)
+{
+    System::Clock::Timestamp currentTime = System::SystemClock().GetMonotonicTimestamp();
+    System::Clock::Timestamp startTime   = PlatformMgrImpl().GetStartTime();
+
+    if (currentTime >= startTime)
+    {
+        upTime = std::chrono::duration_cast<System::Clock::Seconds64>(currentTime - startTime).count();
+        return CHIP_NO_ERROR;
+    }
+
+    return CHIP_ERROR_INVALID_TIME;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetTotalOperationalHours(uint32_t & totalOperationalHours)
+{
+    uint64_t upTimeS;
+
+    ReturnErrorOnFailure(GetUpTime(upTimeS));
+
+    uint32_t totalHours      = 0;
+    const uint32_t upTimeH   = upTimeS / 3600 < UINT32_MAX ? static_cast<uint32_t>(upTimeS / 3600) : UINT32_MAX;
+    const uint32_t deltaTime = upTimeH - PlatformMgrImpl().GetSavedOperationalHoursSinceBoot();
+
+    ReturnErrorOnFailure(ConfigurationMgr().GetTotalOperationalHours(totalHours));
+
+    totalOperationalHours = totalHours + deltaTime < UINT32_MAX ? totalHours + deltaTime : UINT32_MAX;
```

Comment:
totalHours and deltaTime are both `uint32_t`, so the addition here can overflow to a very small number and no one would notice.  This check needs to be different: either cast things to uint64_t, or compare subtracting one from UINT32_MAX to the other, or something like that.

Reply:
Right, thanks for spotting that, I fixed it. Overflowing 32 bit hours counter could take like few thousands of years, but anyway that could be error prone in some fault cases.